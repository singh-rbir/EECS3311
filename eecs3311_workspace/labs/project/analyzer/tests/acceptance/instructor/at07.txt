add_class("A")

add_query("A", "qa", <<>>, "INTEGER")
add_command("A", "ca", <<>>)

-- Error: add implementation to some non-existing routine in class A.
add_assignment_instruction("A", "qb", "x")
add_assignment_instruction("A", "cb", "y")

add_attribute("A", "att", "BOOLEAN")

-- Error: add implementation to an attribute
add_assignment_instruction("A", "att", "z")

add_assignment_instruction("A", "qa", "att")

-- Error: add an empty call chain as assignment RHS
add_call_chain(<<>>)

-- After this, the current assignment instruction `att = ?` is completed (i.e., `att = True`).
bool_value(True)