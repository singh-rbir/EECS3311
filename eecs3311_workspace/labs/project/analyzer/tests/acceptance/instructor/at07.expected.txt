  Status: OK.
  Number of classes being specified: 0
->add_class("A")
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_query("A","qa",<<>>,"INTEGER")
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 1
        + qa: INTEGER
      Number of commands: 0
->add_command("A","ca",<<>>)
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 1
        + qa: INTEGER
      Number of commands: 1
        + ca
->add_assignment_instruction("A","qb","x")
  Status: Error (qb is not an existing feature name in class A).
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 1
        + qa: INTEGER
      Number of commands: 1
        + ca
->add_assignment_instruction("A","cb","y")
  Status: Error (cb is not an existing feature name in class A).
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 1
        + qa: INTEGER
      Number of commands: 1
        + ca
->add_attribute("A","att","BOOLEAN")
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + att: BOOLEAN
      Number of queries: 1
        + qa: INTEGER
      Number of commands: 1
        + ca
->add_assignment_instruction("A","att","z")
  Status: Error (Attribute att in class A cannot be specified with an implementation).
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + att: BOOLEAN
      Number of queries: 1
        + qa: INTEGER
      Number of commands: 1
        + ca
->add_assignment_instruction("A","qa","att")
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + att: BOOLEAN
      Number of queries: 1
        + qa: INTEGER
      Number of commands: 1
        + ca
  Routine currently being implemented: {A}.qa
  Assignment being specified: att := ?
->add_call_chain(<<>>)
  Status: Error (Call chain is empty).
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + att: BOOLEAN
      Number of queries: 1
        + qa: INTEGER
      Number of commands: 1
        + ca
  Routine currently being implemented: {A}.qa
  Assignment being specified: att := ?
->bool_value(True)
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + att: BOOLEAN
      Number of queries: 1
        + qa: INTEGER
      Number of commands: 1
        + ca
